packages/@aws-cdk/custom-resource-handlers/lib/aws-s3/notifications-resource-handler/index.py

```diff
@@ -34,20 +35,31 @@ def handle_managed(request_type, notification_configuration):
     return {}
   return notification_configuration
 
-def handle_unmanaged(bucket, stack_id, request_type, notification_configuration, old):
+def handle_unmanaged(bucket, stack_id, request_type, notification_configuration, old, s3NotificationsDeleteFeatureFlagEnabled):
+  def get_id(n):
+    n['Id'] = ''
+    strToHash=json.dumps(n, sort_keys=True).replace('"Name": "prefix"', '"Name": "Prefix"').replace('"Name": "suffix"', '"Name": "Suffix"')
+    return f"{stack_id}-{hash(strToHash)}"
   def with_id(n):
-    n['Id'] = f"{stack_id}-{hash(json.dumps(n, sort_keys=True))}"
+    if s3NotificationsDeleteFeatureFlagEnabled:
+      n['Id'] = get_id(n)
```

Comment:
Can the `get_id` uses the same naming format where it starts with `stack-id` so the `Delete` event can handle it correctly since it relies on `{stack-id}` prefix.

Reply:
Yes, it's using the same format.

Reply:
Oh yeah, my bad for not noticing it's using the same format. Thanks for addressing this.